modules/core/include/opencv2/core/dualquaternion.inl.hpp

```diff
@@ -158,7 +158,7 @@ inline Quat<T> DualQuat<T>::getRotation(QuatAssumeType assumeUnit) const
 template <typename T>
 inline Vec<T, 3> DualQuat<T>::getTranslation(QuatAssumeType assumeUnit) const
 {
-    Quat<T> trans = 2.0 * (getDualPart() * getRealPart().inv(assumeUnit));
+    Quat<T> trans = T(2.0) * (getDualPart() * getRealPart().inv(assumeUnit));
```

Comment:
Such changes should be backported to 4.x

Reply:
Backported to #21319 and [#3137@contrib](https://github.com/opencv/opencv_contrib/pull/3137)